Terminal.Gui/Views/Button.cs

```diff
@@ -145,7 +145,12 @@ internal void Update ()
 				base.Text = ustring.Make (_leftBracket) + " " + text + " " + ustring.Make (_rightBracket);
 
 			int w = base.Text.RuneCount - (base.Text.Contains (HotKeySpecifier) ? 1 : 0);
-			Width = w;
+			try {
```

Comment:
Could this be implemented with an `if "DimCombine"` instead of a `try/catch`? Would this require us to add an API to PosDim to be able to test it ahead of time?

Reply:
I made a more elegant solution. Otherwise you get the button undocked and it gets ugly.

Reply:
> 
> 
> Could this be implemented with an `if "DimCombine"` instead of a `try/catch`? Would this require us to add an API to PosDim to be able to test it ahead of time?

For that we will have to make the class public. Can be?

Reply:
To be clear, I feel pretty strongly that using `try/catch` for anything other than dealing with actual errors is a very bad practice. Can we modify `PosDim` to provide an API such that an `if` statement could be used here?

Reply:
> > Could this be implemented with an `if "DimCombine"` instead of a `try/catch`? Would this require us to add an API to PosDim to be able to test it ahead of time?
> 
> For that we will have to make the class public. Can be?

To me that would be far preferable to a hack with `try/catch`. Which class are we talking about here?

Reply:
Yes, it is possible. I also preferred this to using `try / catch`. I'll take care of that.

Reply:
This one:

`class DimCombine : Dim {`

Reply:
Done!